cinema/tests/test_movie_api.py

```diff
@@ -157,3 +158,156 @@ def test_image_url_is_shown_on_movie_session_detail(self):
+            email="test@test.com",
+            password="testpassword"
+        )
+        self.client = APIClient()
+        self.client.force_authenticate(self.user)
+
+    def test_retrieve_movie_list_endpoint(self):
+        """Test retrieving movies"""
+        sample_movie(title="Movie 1")
+        sample_movie(title="Movie 2")
+
+        res = self.client.get(MOVIE_URL)
+
+        movies = Movie.objects.all().order_by("title")
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_retrieve_movie_list_endpoint_with_filtered_by_title(self):
+        """Test retrieving movies filtered by title"""
+        sample_movie(title="Movie 1")
+        sample_movie(title="Movie 2")
+
+        res = self.client.get(MOVIE_URL, data={"title": "Movie 1"})
+
+        movies = Movie.objects.filter(title="Movie 1")
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_retrieve_movie_list_endpoint_with_filtered_by_genre(self):
+        """Test retrieving movies filtered by genre"""
+        sample_movie(title="Movie 1")
+        sample_movie(title="Movie 2")
+        sample_movie(title="Movie 3")
+
+        res = self.client.get(MOVIE_URL, data={"genres": 1})
+
+        movies = Movie.objects.filter(genres__id=1)
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_retrieve_movie_list_endpoint_with_filtered_by_actor(self):
+        """Test retrieving movies filtered by actor"""
+        sample_movie(title="Movie 1")
+        sample_movie(title="Movie 2")
+        sample_movie(title="Movie 3")
+
+        res = self.client.get(MOVIE_URL, data={"actors": 1})
+
+        movies = Movie.objects.filter(actors__id=1)
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_retrieve_movie_detail_endpoint(self):
+        """Test retrieving movie detail"""
+        movie = sample_movie(title="Movie 1")
+
+        res = self.client.get(detail_url(movie.id))
+
+        serializer = MovieDetailSerializer(movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertEqual(res.data, serializer.data)
+
+    def test_create_movie_forbidden(self):
+        """Test creating a movie is forbidden"""
+        payload = {
+            "title": "Movie 1",
+            "description": "Test description 1",
+            "duration": 80,
+        }
+        res = self.client.post(MOVIE_URL, payload)
+
+        self.assertEqual(res.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
The `test_create_movie_forbidden` test expects a `403 FORBIDDEN` status. Verify that the API is configured to prevent authenticated users from creating movies if this is the intended behavior.